text/0039-reduce-module-size.md

```diff
@@ -0,0 +1,583 @@
+   with that goal by doubling the amount of content that is published on each
+   release. And, there is currently no demonstrated customer need for a debug
+   package like this.
+
+#### Lambda Layer Zip Files
+
+1. Host the zip files in a static website. This would be implemented as a CDK
+   app in a new GitHub repository. The CDK CLI will download and cache these zip
+   files at synth time. The existing `lambda-layer-X` modules will be modified
+   to reference the large dependencies from a local cache. The CDK CLI will be
+   modified to update that cache at synth time, if needed. Then, the rest of
+   their usage will remain the same. The URLs for each dependency would follow a
+   predictable format, including each dependency’s version. Some automated
+   mechanism would be required to add new artifacts to the website as new
+   versions of each dependency are released. And, to add new versions of each
+   dependency to aws-cdk-lib. The problem with this solution is that it requires
+   access to a public endpoint during synthesis of CDK applications and does not
+   lend itself to a straightforward workaround for customers who run CDK apps in
+   network-restricted environments (Construct Hub, ADC region customers, etc).
+2. Download the the zip files from a static website in the framework
+   implementation of each `Layer` class, instead of in the CLI. We should not do
+   this, because the CLI already has logic for managing credentials and making
+   network calls. This type of work should remain in the CLI, and not overflow
+   into the framework.
+3. Use regionalized S3 buckets to host the zip files. With this solution, the
+   customer does not need to download the zip file and then re-upload it to
+   their deployment environment. Instead, the Lambda Layer or Lambda Function
+   resource definition in their CloudFormation template references the zip
+   hosted by us in the same region as their deployment environment. The
+   maintenance burden of keeping up with deploying these artifacts and keeping
+   them up to date in every AWS region is too high. This would effectively make
+   our team double as a hosted artifact repository for artifacts that we don't
+   own. This solution is the same as the one described in this
+   [comment](https://github.com/aws/aws-cdk-rfcs/issues/39#issuecomment-593092612).
+4. Separate packages that `aws-cdk-lib` peer depends on. With this solution,
+   customers do not need to include the large `lambda-layer-X` packages in their
+   dependencies unless they are actually using one of the Constructs that
+   requires them. This would be confusing for customers to know when they need
+   these peer dependencies, and when they do not. It would also be difficult for
+   customers to manage versions and to know which versions of each package are
+   compatible with each other.
+5. Upstream the custom resources where the zip files are used. Since all of
```

Comment:
A similar approach that suffers from them same problem would have been to setup an internal pipeline that publishes AWS provided layers, such as:

![Screen Shot 2022-07-17 at 3 57 08 PM](https://user-images.githubusercontent.com/1428812/179399436-168d5fe2-9947-4b83-9851-c48c6a39cef1.png)

Thought I'd put it on here anyway for posterity. Also - it might be worth while investigating how exactly this later is published.

Reply:
Good callout, I did look into this approach. It basically has the same drawbacks as an internal pipeline publishing to s3. I heard from someone, but did not verify myself, that with Lambda Layers, you publish under your own account id, and so it is not possible to use convention-based naming.

Reply:
I added this solution to the list of alternatives.